types/pointer/pointer.ts

```diff
@@ -0,0 +1,31 @@
+import { AlignedType } from "../types.ts";
+import { endianess } from "../../utils.ts";
+
+export class Pointer<T> implements AlignedType<T> {
+  byteLength = 8;
+  byteAlign = 8;
```

Comment:
I'm not sure if this is correct. Cuz on 32 bit platforms the ptr size is 4 bytes and aligns with 4 rather than 8 (not sure if this is even relevant if deno does not have any 32 bit compile targets)

Reply:
Hmm, good point. I think it could be wise to instead wrap one of the integer types and then create the actual pointer from that type (e.g. any `AlignedType<number | bigint>`). That way we can either let the user decide the pointer width or automatically choose `u32`/`u64` depending on the platform (or just choose `u64` by default).

Reply:
That was what I was gonna do initially but I figured it was unnecessary since Deno doesn't support 32bit platforms

Reply:
Sure, good point but it could be useful for other weird non-standard pointer representations (packed pointers maybe? or on arm platforms?).